core/src/memory.rs

```diff
@@ -376,15 +379,29 @@ impl AllocationsContext {
         Ok(start)
     }
 
-    /// Free specific page.
-    ///
-    /// Currently running program should own this page.
+    /// Free specific memory page.
     pub fn free(&mut self, page: WasmPage) -> Result<(), AllocError> {
-        if page < self.static_pages || page >= self.max_pages || !self.allocations.remove(&page) {
-            Err(AllocError::InvalidFree(page.0))
-        } else {
-            Ok(())
+        if page < self.static_pages || page >= self.max_pages {
```

Comment:
wasn't previous extra || more compact?

Reply:
more compact, yes, but it was hard to comprehend mutation in `if` condition